app/addons/index.js

```diff
@@ -1,78 +1,339 @@
-//internal modules
-var fs = require('fs');
-var path = require('path');
-
-//3rd party modules
-var _ = require('lodash')
-var logger = require('winston');
-var async = require('async');
-
-function AddonManager (app, server, io){
-  var self = this;
-  var addons = [];
-
-  this.RegisterAddons = function() {
-    logger.info("Loading addons..");
-    fs.readdirSync(__dirname).forEach(function (file) {
-      if (!file.match(/\.js$/) && !file.match(/^\./) ) {
-         logger.verbose(" * "+file);
-         var addonPath = path.join(__dirname, file)
-         try {
-           let packageJsonFile = path.join(addonPath, 'package.json');
-           let packageJson = require(packageJsonFile);
-           let deps = Object.keys(_.get(packageJson, 'dependencies', {}));
-           _.each(deps, (dep) => {
-              try {
-                  require.resolve(dep);
-              } catch(e) {
-                  logger.warn(dep + " npm package is not found, required by addon "+file);
-                  deps = false;
-              }
-           });
-           if(deps === false) {
-             return;
-           }
-         } catch(e){
-           logger.debug(e);
-         }
-         try {
-           var Addon = require(addonPath);
-           if(Addon.disabled) {
-             logger.info('Addon %s is disabled', file);
-             return;
-           }
-           var addon = new Addon(app, server, io);
-           addon.register();
-           addons.push( addon  );
-         } catch(e) {
-            addonPath = path.relative('.', addonPath)
-            logger.error('Cannot load addon "%s": %s', addonPath, e.toString());
-            logger.debug(e.stack);
-         }
-      }
+// native modules
+const fs = require('fs');
+const path = require('path');
+
+// 3rd party modules
+const logger = require('winston');
+
+process.on('warning', (warn) => {
+  console.log(warn.stack);
+});
+
+const STATES = { introduce: 0, load: 1, instantiate: 2, register: 3, unregister: 4 };
+const PHASES = { inProgress: 0, done: 1, failed: 2 };
+
+function createErrorMessage(pError, pMetaInfo, pSolution) {
+  let combinedError = pError;
+
+  // Append meta data lines
+  for (let i = 0; i < pMetaInfo.length; i += 1) {
```

Comment:
perhaps could be even more simple like: `_.for(pMetaInfo, (info) => { combinedError+=...}`